src/dotnet/GeneratorAgent/src/Azure.Tools.GeneratorAgent/Agent/AgentResponseParser.cs

```diff
@@ -0,0 +1,193 @@
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Parses the agent's error responses into a list of RuleError objects
+        /// </summary>
+        public List<RuleError> ParseErrors(string rawResponse)
+        {
+            Logger.LogInformation("Parsing errors from raw agent response.");
+
+            try
+            {
+                if (string.IsNullOrWhiteSpace(rawResponse))
+                {
+                    throw new InvalidOperationException("Agent response is empty or null");
+                }
+
+                // Extract JSON from markdown code blocks if present
+                var jsonContent = ExtractJsonFromResponse(rawResponse);
+
+                // Deserialize the response
+                var response = JsonSerializer.Deserialize<AgentErrorResponse>(jsonContent, new JsonSerializerOptions
+                {
+                    PropertyNameCaseInsensitive = true,
+                    AllowTrailingCommas = true
+                });
+
+                if (response == null || response.Errors == null || !response.Errors.Any())
+                {
+                    throw new InvalidOperationException("No errors found in the agent response");
+                }
+
+                // Map errors to RuleError objects
+                var ruleErrors = response.Errors.Select(e => new RuleError(e.Type, e.Message)).ToList();
+
+                Logger.LogInformation("Successfully parsed {Count} errors from agent response.", ruleErrors.Count);
+                return ruleErrors;
+            }
+            catch (JsonException ex)
+            {
+                Logger.LogError(ex, "Failed to parse agent response as JSON. Response: {Response}", rawResponse);
+                throw new InvalidOperationException("Agent response is not in the expected JSON format", ex);
+            }
+            catch (Exception ex)
+            {
+                Logger.LogError(ex, "Unexpected error while parsing agent response. Response length: {Length}", rawResponse?.Length ?? 0);
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Extracts JSON content from agent response, handling markdown code blocks if present
+        /// </summary>
+        private string ExtractJsonFromResponse(string rawResponse)
+        {
+            if (string.IsNullOrWhiteSpace(rawResponse))
+            {
+                return rawResponse;
+            }
+
+            var trimmed = rawResponse.Trim();
+            
+            // Check if response is wrapped in markdown code blocks
+            if (trimmed.StartsWith("```json", StringComparison.OrdinalIgnoreCase))
+            {
+                // Find the start of JSON (after ```json)
+                var startIndex = trimmed.IndexOf('\n');
+                if (startIndex == -1)
+                {
+                    Logger.LogWarning("Malformed markdown code block: no newline after ```json");
```

Comment:
In regards to the above, I think since this is a response that's really out of the control of the user, just sort of deciding which outcomes are malformed enough that it's concerning, throwing in those cases, otherwise just using what we got and not worrying about logs should work ok?